plugins/colorschemes/kanagawa-paper.nix

```diff
@@ -0,0 +1,230 @@
+  '';
+
+  maintainers = [ lib.maintainers.FKouhai ];
+
+  # TODO introduced 2024-03-15: remove 2024-05-15
+  deprecateExtraOptions = true;
+  optionsRenamedToSettings = [
+    "compile"
+    "undercurl"
+    "comments"
+    "functions"
+    "keyword"
+    "statement"
+    "type"
+    "transparent"
+    "dim_inactive"
+    "terminal_colors"
+    [
+      "colors"
+      "palette"
+    ]
+    [
+      "colors"
+      "theme"
+    ]
+    "theme"
+    [
+      "background"
+      "dark"
+    ]
+    [
+      "background"
+      "light"
+    ]
+
+  ];
+  imports = [
+    (lib.mkRemovedOptionModule
+      [
+        "colorschemes"
+        "kanagawa-paper"
+        "overrides"
+      ]
+      ''
+        Use `colorschemes.kanagawa-paper.settings.overrides` but you now have to add the full function definition:
+        ```
+          function(colors)
+            ...
+          end
+        ```
+      ''
+    )
+  ];
+
+  settingsOptions = {
+    compile = defaultNullOpts.mkBool false ''
+      Enable compiling the colorscheme.
+    '';
+
+    undercurl = defaultNullOpts.mkBool true ''
+      Enable undercurls.
+    '';
+
+    comments = defaultNullOpts.mkAttrsOf types.anything { italic = true; } ''
+      Highlight options for comments.
+    '';
+
+    function = defaultNullOpts.mkAttrsOf types.anything { } ''
+      Highlight options for functions.
+    '';
+
+    keyword = defaultNullOpts.mkAttrsOf types.anything { italic = true; } ''
+      Highlight options for keywords.
+    '';
+
+    statement = defaultNullOpts.mkAttrsOf types.anything { bold = true; } ''
+      Highlight options for statements.
+    '';
+
+    type = defaultNullOpts.mkAttrsOf types.anything { } ''
+      Highlight options for types.
+    '';
+
+    transparent = defaultNullOpts.mkBool false ''
+      Whether to set a background color.
+    '';
+
+    dim_inactive = defaultNullOpts.mkBool false ''
+      Whether dim inactive window `:h hl-NormalNC`.
+    '';
+
+    terminal_colors = defaultNullOpts.mkBool true ''
+      If true, defines `vim.g.terminal_color_{0,17}`.
+    '';
+
+    colors = {
+      theme =
+        defaultNullOpts.mkAttrsOf types.attrs
+          {
+            ink = { };
+            canvas = { };
+            all = { };
+          }
+          ''
+            Change specific usages for a certain theme, or for all of them
+
+            Example:
+            ```nix
+              {
+                ink = {
+                  ui = {
+                      float = {
+                          bg = "none";
+                      };
+                  };
+                };
+                canvas = {
+                    syn = {
+                        parameter = "yellow";
+                    };
+                };
+                all = {
+                    ui = {
+                        bg_gutter = "none";
+                    };
+                };
+              }
+            ```
```

Comment:
We usually avoid manually putting examples in the description. We should use `mkOption`'s `example` arg.

To get access to passing mkOption args in a defaultNullOpts function, use the `'` (prime) variant.

---

That said, we've been gradually transitioning away from declaring any `settingsOptions` _at all_. Preferring to _only_ define `settingsExample`.